Common/Data/Auxiliary/FactorFileRow.cs

```diff
@@ -180,15 +180,15 @@ public FactorFileRow Apply(Split split, SecurityExchangeHours exchangeHours)
                 throw new ArgumentException("Unable to apply split with reference price of zero.");
             }
 
+            var previousTradingDay = exchangeHours.GetPreviousTradingDay(split.Time);
+
             // this instance must be chronologically at or in front of the split
             // this is because the factors are defined working from current to past
-            if (Date < split.Time.Date)
+            if (Date < previousTradingDay)
```

Comment:
Was this a bug fix? Curious where the motivation for it came from?

Reply:
the date in the factor file is the trading day _before_ the split/dividend event. I found this when writing a test to apply a split/dividend on the same day.